<launch>

  <node pkg="giskard_examples" type="pr2_controller" name="pr2_controller" output="screen">
    <rosparam param="joint_names"
      file="$(find giskard_examples)/config/pr2_left_upper_body.yaml" />
    <param name="controller_description" 
      textfile="$(find giskard_examples)/controller_specs/pr2_left_upper_body_qp_position_control.yaml" />
    <param name="frame_id" value="base_link" />
    <remap from="~joint_states" to="/joint_states" />
  </node>

  <node pkg="giskard_examples" type="whole_body_interactive_markers" 
        name="whole_body_interactive_markers" output="screen">
    <param name="left_ee_frame_id" value="l_gripper_tool_frame" />
    <param name="left_ee_name" value="LEFT GRIPPER" />
    <param name="right_ee_frame_id" value="r_gripper_tool_frame" />
    <param name="right_ee_name" value="RIGHT GRIPPER" />
    <remap from="~goal_out" to="/goal_transformer/in" />
  </node>

  <node pkg="giskard_examples" type="goal_transformer" 
        name="goal_transformer" output="screen">
    <param name="frame_id" value="base_link" />
    <remap from="~out" to="/pr2_controller/goal" />
  </node>

</launch>
